File: ClassLibrary/Enums/Urls.cs
namespace ClassLibrary.Enums
{
    public class Urls
    {
        /* QUARX AUTH */
        private static readonly string AuthClientProtocol = "http";
        private static readonly string AuthClientHost = "192.168.1.81";
        public static readonly string AuthClientPort = "1810";

        private static readonly string AuthServerProtocol = "http";
        private static readonly string AuthServerHost = "192.168.1.81";
        private static readonly string AuthServerPort = "1910";


        /* HRTECHNIQUE */
        private static readonly string HrtechniqueClientProtocol = "http";
        private static readonly string HrtechniqueClientHost = "192.168.1.81";
        private static readonly string HrtechniqueClientPort = "1830";

        private static readonly string HrtechniqueServerProtocol = "http";
        private static readonly string HrtechniqueServerHost = "192.168.1.81";
        private static readonly string HrtechniqueServerPort = "1930";



        public enum Name
        {
            AuthClient,
            AuthServer,
            HrTechniqueClient,
            HrTechniqueServer,
        }

        public static string GetAssociatedUrl(Name name)
        {
            switch (name)
            {
                case Name.AuthClient:
                    return $"{AuthClientProtocol}://{AuthClientHost}:{AuthClientPort}";
                case Name.AuthServer:
                    return $"{AuthServerProtocol}://{AuthServerHost}:{AuthServerPort}";
                case Name.HrTechniqueClient:
                    return $"{HrtechniqueClientProtocol}://{HrtechniqueClientHost}:{HrtechniqueClientPort}";
                case Name.HrTechniqueServer:
                    return $"{HrtechniqueServerProtocol}://{HrtechniqueServerHost}:{HrtechniqueServerPort}";
                default:
                    return string.Empty;
            }
        }
    }
}

File: AuthClient/src/variables/urls.ts
export enum Urls {
  /* HRT AUTH */
  HOST = '192.168.1.77',
  AUTH_CLIENT_PROTOCOL = 'http',
  AUTH_CLIENT_HOST = '192.168.1.77',
  AUTH_CLIENT_PORT = '1810',

  AUTH_SERVER_PROTOCOL = 'http',
  AUTH_SERVER_HOST = '192.168.1.77',
  AUTH_SERVER_PORT = '1910',


  /* HRTECHNIQUE */
  HRTECHNIQUE_CLIENT_PROTOCOL = 'http',
  HRTECHNIQUE_CLIENT_HOST = '192.168.1.77',
  HRTECHNIQUE_CLIENT_PORT = '1830',

  HRTECHNIQUE_SERVER_PROTOCOL = 'http',
  HRTECHNIQUE_SERVER_HOST = '192.168.1.77',
  HRTECHNIQUE_SERVER_PORT = '1930',

  GITHUB_REPO = 'https://github.com/vuelcomponents/QIVI-SYSTEMS'
}
export type Name =
  | 'auth-client'
  | 'auth-server'
  | 'hrtechnique-client'
  | 'hrtechnique-server'
  ;
export const getAssociatedUrl = (name: Name): string => {
  switch (name) {
    case 'auth-client':
      return `${Urls.AUTH_CLIENT_PROTOCOL}://${Urls.AUTH_CLIENT_HOST}:${Urls.AUTH_CLIENT_PORT}`;
    case 'auth-server':
      return `${Urls.AUTH_SERVER_PROTOCOL}://${Urls.AUTH_SERVER_HOST}:${Urls.AUTH_SERVER_PORT}`;
    case 'hrtechnique-client':
      return `${Urls.HRTECHNIQUE_CLIENT_PROTOCOL}://${Urls.HRTECHNIQUE_CLIENT_HOST}:${Urls.HRTECHNIQUE_CLIENT_PORT}`;
    case 'hrtechnique-server':
      return `${Urls.HRTECHNIQUE_SERVER_PROTOCOL}://${Urls.HRTECHNIQUE_SERVER_HOST}:${Urls.HRTECHNIQUE_SERVER_PORT}`;
 }
};
